distgen.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -e
+
+# before using this script, make sure your image repository contains:
+#   ./templates directory
+#   ./specs/multispec.yml file
+#   ./dg_manifest file
+
+# All templates should be stored in templates directory in container repository
+if [ ! -d "./templates" ]; then
+    echo "Templates for distgen not found"
+    exit
+fi
+
+# expecting file dg_manifest that contains variables:
+#   FILES_TO_COPY in form "path1:mode1 path2:mode2..."
+#   FILES_TO_GEN in form similar to FILES_TO_COPY
+#   INTERNAL_SYMLINKS in form "name1->target1 name1->target1..."
+if [ -f "./dg_manifest" ]; then
+    source ./dg_manifest
+fi
+
+WORKDIR=${PWD}
+DG=${DG:-/usr/bin/dg}
+DG_CONF=${DG_CONF:-centos-7-x86_64.yaml}
+
+cd templates
+OUTPUT_DIR=$(mktemp -d ${WORKDIR}/${OS}-${VERSION}-XXXX)
+echo "Copying files to ${OUTPUT_DIR}:"
+for file in ${FILES_TO_COPY}; do
```

Comment:
I would really prefer having this solved properly through Makefile targets, so
everything is fully parallelized (even this cp && chmod thingy, but especially
the $FILES_TO_GEN bellow).

Thanks for the chmod call btw.!  Could we made that "optional" with some sane
default (say 644)?  Btw., this could solve #3 pretty nicely :-).